src/util/file-downloader.ts

```diff
@@ -0,0 +1,69 @@
+import { IHttpClient } from '../sdk/http';
+import { IDownloadFileProvider } from '../sdk/resources';
+
+const path = require('path');
+const fs = require('fs');
+
+export class FileDownloader {
+	constructor(private httpClient: IHttpClient, private fileProvider: IDownloadFileProvider) {}
+
+	createDirectory(targetDirectory: string) {
```

Comment:
`private` or pull it up as a function outside of this class so that it doesn't get exported with the public contract.